lms/validation/_launch_params.py

```diff
@@ -0,0 +1,80 @@
+"""Schema for validating LTI launch params."""
+import marshmallow
+from webargs.pyramidparser import parser
+from pyramid.httpexceptions import HTTPUnprocessableEntity
+
+from lms.services import LTILaunchVerificationError
+from lms.validation._exceptions import ValidationError
+from lms.values import LTIUser
+
+
+__all__ = ("LaunchParamsSchema",)
+
+
+class LaunchParamsSchema(marshmallow.Schema):
+    """
+    Schema for LTI launch params.
+
+    Validates and verifies LTI launch params. Usage via webargs::
+
+        >>> from webargs.pyramidparser import parser
+        >>>
+        >>> schema = LaunchParamsSchema(request)
+        >>> parsed_params = parser.parse(schema, request, locations=["form"])
+
+    Or to verify the request and get an :class:`~lms.values.LTIUser`
+    from the request's params::
+
+        >>> schema = LaunchParamsSchema(request)
+        >>> schema.lti_user()
+        LTIUser(user_id='...', ...)
+    """
+
+    user_id = marshmallow.fields.Str(required=True)
+
+    oauth_consumer_key = marshmallow.fields.Str(required=True)
+    oauth_nonce = marshmallow.fields.Str(required=True)
+    oauth_signature = marshmallow.fields.Str(required=True)
+    oauth_signature_method = marshmallow.fields.Str(required=True)
+    oauth_timestamp = marshmallow.fields.Str(required=True)
+    oauth_version = marshmallow.fields.Str(required=True)
```

Comment:
All of these OAuth 1 fields are required for the OAuth 1 signature verification to pass.